spec.md

```diff
@@ -248,13 +248,13 @@ service Controller {
     returns (CreateVolumeResponse) {}
 
   rpc DeleteVolume (DeleteVolumeRequest)
-    returns (DeleteVolumeResponse) {}
+    returns (Empty) {}
```

Comment:
I'm conflicted on this. While I agree with much of @jieyu's reasoning, this bit strikes me in a funny way:

> add an optional field to `DeleteVolumeResponse` will be backward compatible (without the need for bumping a CSI version).

While this is technically *true*, if there *is* additional information added to otherwise empty responses, is bumping the version such a bad thing? Otherwise the version wouldn't reflect the addition of new information. Let's look at the stated behavior of SemVer:

> MAJOR version when you make incompatible API changes,
> MINOR version when you add functionality in a backwards-compatible manner, and
> PATCH version when you make backwards-compatible bug fixes.

It would seem then that a MINOR version change would be required in the case of keeping `DeleteVolumeResponse` for now in order to possibly add new data to it in the future. However, I'm not sure I agree with this. To me a MINOR change is valid if we, for example, added a new RPC that provides extended functionality. However, changing the response data for an RPC, even if technically backwards compatible, is to me a MAJOR change as it affects an existing RPC. Again, I understand it is technically a MINOR change, but I'm talking about in practice, not in theory.

Therefore I'm fine leaving this items `Empty` for now and then requiring a MAJOR version bump if we should ever change them.

Reply:
This would only be backwards incompatible at the library layer, right?
The proto/wire format should be fine.

Reply:
Yeah, i think wire format should be the same. Just i feel the benefit of doing this is thin. It's nice to keep request/response symmetric.

Reply:
> This would only be backwards incompatible at the library layer, right?

That's correct. Truth be told though, adding a new RPC, while a MINOR SemVer bump, would be backwards incompatible for any Go library.

I think we have to agree (or not) that compatibility is with regards to the spec, not with regards to a specific language binding/implementation of the spec.

Reply:
> I think we have to agree (or not) that compatibility is with regards to the spec, not with regards to a specific language binding/implementation of the spec.

Actually, I think the above is a good topic for next week's meeting. Is Go prevalent enough that we should consider it when considering SemVer for the spec?

Reply:
I don't feel strongly either way. Since both @jieyu strongly prefers symmetry, I'll go ahead and make it symmetric.